src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -49,7 +52,12 @@ public Connection(ListenerContext context, UvStreamHandle socket) : base(context
 
             ConnectionId = GenerateConnectionId(Interlocked.Increment(ref _lastConnectionId));
 
-            _rawSocketInput = new SocketInput(Memory, ThreadPool);
+            if (ServerOptions.MaxInputBufferLength.HasValue)
```

Comment:
I'm not sure how the default behaves  `private int? _maxInputBufferLength = 1024 * 1024; `
Does this mean 1MB is the default and in that case will HasValue every be false?

Basically is there a scenario where we never use the BufferLenghtControl which would cause large post to kill the server.